crates/biome_service/src/workspace/scanner.rs

```diff
@@ -211,6 +213,35 @@ pub(crate) struct ScanContext<'app> {
 
     /// List of paths that should be processed.
     pub(crate) evaluated_paths: RwLock<BTreeSet<BiomePath>>,
+
+    /// What the scanner should target.
+    target: ScanKind,
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase")]
+pub enum ScanKind {
+    /// It targets the project, so it attempts to open all the files in the project.
+    Project,
+    /// It targets specific files
+    KnownFiles,
+    /// The scanner should not be triggered
+    None,
+}
+
+impl ScanKind {
+    pub const fn is_project(&self) -> bool {
+        matches!(self, ScanKind::Project)
+    }
+
+    pub const fn is_known_files(&self) -> bool {
+        matches!(self, ScanKind::KnownFiles)
+    }
+
+    pub const fn is_none(&self) -> bool {
+        matches!(self, ScanKind::None)
+    }
 }
```

Comment:
I suggest deriving Copy and taking ownership in methods.
```suggestion
#[derive(Debug, Copy, Clone, serde::Serialize, serde::Deserialize)]
#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
#[serde(rename_all = "camelCase")]
pub enum ScanKind {
    /// It targets the project, so it attempts to open all the files in the project.
    Project,
    /// It targets specific files
    KnownFiles,
    /// The scanner should not be triggered
    None,
}

impl ScanKind {
    pub const fn is_project(self) -> bool {
        matches!(self, ScanKind::Project)
    }

    pub const fn is_known_files(self) -> bool {
        matches!(self, ScanKind::KnownFiles)
    }

    pub const fn is_none(self) -> bool {
        matches!(self, ScanKind::None)
    }
}
```